eng/pipelines/templates/jobs/ci.yml

```diff
@@ -226,17 +228,39 @@ jobs:
           BaselineFilePath: $(Build.sourcesdirectory)/eng/dotnet.gdnbaselines
           ServiceDirectory: ${{ parameters.ServiceDirectory }}
 
-  - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
-    parameters:
-      JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
-      MatrixConfigs: ${{ parameters.MatrixConfigs }}
-      MatrixFilters: ${{ parameters.MatrixFilters }}
-      MatrixReplace: ${{ parameters.MatrixReplace }}
-      CloudConfig:
-        Cloud: Public
-      AdditionalParameters:
-        SDKType: ${{ parameters.SDKType }}
-        ServiceToTest: ${{ coalesce(parameters.ServiceToTest, parameters.ServiceDirectory) }}
-        ServiceDirectory: ${{ parameters.ServiceDirectory }}
-        TestSetupSteps: ${{ parameters.TestSetupSteps }}
-        TestTimeoutInMinutes: ${{ parameters.TestTimeoutInMinutes }}
+        ${{ if ne(parameters.TestDependsOnDependency, 'not-specified') }}:
+          - template: /eng/common/pipelines/templates/jobs/archetype-sdk-tests-generate.yml
+            parameters:
+              JobTemplatePath: /eng/pipelines/templates/jobs/ci.tests.yml
+              MatrixConfigs: ${{ parameters.MatrixConfigs }}
+              MatrixFilters: ${{ parameters.MatrixFilters }}
+              MatrixReplace: ${{ parameters.MatrixReplace }}
+              CloudConfig:
+                Cloud: public
+              SparseCheckoutPaths: 
+                - "!/sdk/*/**/SessionRecords"
+              PreGenerationSteps: 
```

Comment:
I wonder if it's simpler to not duplicate the whole step but rather hide this section and the project list parameter behind the conditional instead.